x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/query/ShapeQueryPointProcessor.java

```diff
@@ -8,163 +8,50 @@
 
 import org.apache.lucene.document.XYDocValuesField;
 import org.apache.lucene.document.XYPointField;
-import org.apache.lucene.geo.XYCircle;
-import org.apache.lucene.geo.XYRectangle;
-import org.apache.lucene.search.BooleanClause;
-import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.geo.XYGeometry;
 import org.apache.lucene.search.IndexOrDocValuesQuery;
 import org.apache.lucene.search.Query;
+import org.elasticsearch.common.geo.LuceneGeometriesUtils;
 import org.elasticsearch.common.geo.ShapeRelation;
-import org.elasticsearch.geometry.Circle;
 import org.elasticsearch.geometry.Geometry;
-import org.elasticsearch.geometry.GeometryCollection;
-import org.elasticsearch.geometry.GeometryVisitor;
-import org.elasticsearch.geometry.LinearRing;
-import org.elasticsearch.geometry.MultiLine;
-import org.elasticsearch.geometry.MultiPoint;
-import org.elasticsearch.geometry.MultiPolygon;
-import org.elasticsearch.geometry.Point;
-import org.elasticsearch.geometry.Polygon;
-import org.elasticsearch.geometry.Rectangle;
 import org.elasticsearch.geometry.ShapeType;
 import org.elasticsearch.index.mapper.MappedFieldType;
 import org.elasticsearch.index.query.QueryShardException;
 import org.elasticsearch.index.query.SearchExecutionContext;
-import org.elasticsearch.xpack.spatial.common.ShapeUtils;
 import org.elasticsearch.xpack.spatial.index.mapper.PointFieldMapper;
 
+import java.util.function.Consumer;
+
 public class ShapeQueryPointProcessor {
 
-    public Query shapeQuery(Geometry shape, String fieldName, ShapeRelation relation, SearchExecutionContext context) {
-        validateIsPointFieldType(fieldName, context);
+    public Query shapeQuery(Geometry geometry, String fieldName, ShapeRelation relation, SearchExecutionContext context) {
+        final boolean hasDocValues = validateIsPointFieldType(fieldName, context);
         // only the intersects relation is supported for indexed cartesian point types
         if (relation != ShapeRelation.INTERSECTS) {
             throw new QueryShardException(context, relation + " query relation not supported for Field [" + fieldName + "].");
         }
-        // wrap XYPoint query as a ConstantScoreQuery
-        return getVectorQueryFromShape(shape, fieldName, relation, context);
+        final Consumer<ShapeType> checker = t -> {
+            if (t == ShapeType.POINT || t == ShapeType.MULTIPOINT || t == ShapeType.LINESTRING || t == ShapeType.MULTILINESTRING) {
+                throw new QueryShardException(context, "Field [" + fieldName + "] does not support " + t + " queries");
+            }
+        };
+        final XYGeometry[] luceneGeometries = LuceneGeometriesUtils.toXYGeometry(geometry, checker);
+        Query query = XYPointField.newGeometryQuery(fieldName, luceneGeometries);
```

Comment:
Am I understanding this correctly? The old code would generate one queryu for one of Circle, Polygon, MultiPolygon and Rectangle, while the new code would generate a single query containing all of those types? Or perhaps GeometryCollection is not supported so we will never see more than one anyway?